docs/developer/architecture/authentication.rst

```diff
@@ -0,0 +1,71 @@
+Authentication Architecture Document
+====================================
+
+Context:
+--------
+Currently, when we deploy MetalK8s we pre provision super admin user with admin
+password. Meaning anyone wanting to use K8S/Salt APIs should authenticate using
+this single super admin user.
+This is not good enough in term of security and flexibility.
+
+We would like at least to be able to add different users with different
+credentials and ideally to integrate K8S authentication system with external
+identity provider.
+
+Managing role binding between user / groups and roles is not part of this
+specification.
+
+Requirements:
+-------------
+
+Basically we are talking about
+
+- Being able to provision users with an embedded identity provider
+- Being able to integrate with an external Identity provider
+
+Dex https://github.com/dexidp/dex seems to be the preferred solution for now,
+for both local identity provider or proxy to external identity provider.
+Integration with LDAP or Microsoft AD are the most important ones to support.
+
+We may want to keep the concept of pre provisioned superadmin user
+whatever Identity provider method we use ... to be discussed. 
```

Comment:
Agree that we might need to have a statically pre-defined Dex superuser. If we intend to support some sort of user management(add/delete) in the Metalk8s UI someday, then this is mandatory since User addition over the Dex GRPC API would only work when the user performing the action has admin privileges.